metadata-ingestion/src/datahub/ingestion/source/sql/mysql.py

```diff
@@ -219,30 +332,40 @@ def get_procedures_for_schema(
         base_procedures = []
         with inspector.engine.connect() as conn:
             procedures = conn.execute(
-                """
-                SELECT ROUTINE_NAME AS name, 
-                    ROUTINE_DEFINITION AS definition, 
-                    EXTERNAL_LANGUAGE AS language
-                FROM information_schema.ROUTINES
-                WHERE ROUTINE_TYPE = 'PROCEDURE'
-                AND ROUTINE_SCHEMA = %s
-                """,
-                (schema,),
+                text(STORED_PROCEDURES_QUERY),
+                {"schema": schema},
             )
 
-            procedure_rows = list(procedures)
+            procedure_rows: List[Row] = list(procedures)
             for row in procedure_rows:
+                # Convert SQLAlchemy Row to dict for easier and safer access
+                row_dict = dict(row)
+
+                # Extract name - name is required for BaseProcedure
+                procedure_name = row_dict.get("name")
+                if not procedure_name:
+                    # Skip procedures without names
+                    continue
+
                 base_procedures.append(
                     BaseProcedure(
-                        name=row.name,
-                        language=row.language,
+                        name=procedure_name,
+                        language="SQL",
                         argument_signature=None,
                         return_type=None,
-                        procedure_definition=row.definition,
-                        created=None,
-                        last_altered=None,
-                        extra_properties=None,
-                        comment=None,
+                        procedure_definition=row_dict.get("definition"),
```

Comment:
- MySQL (mysql.py:357-358): Passes datetime fields directly from the query
  created=row_dict.get("CREATED"),
  last_altered=row_dict.get("LAST_ALTERED"),
  - MariaDB (mariadb.py:94-95): Parses them with _parse_datetime()
  created=self._parse_datetime(row_dict.get("CREATED")),
  last_altered=self._parse_datetime(row_dict.get("LAST_ALTERED")),

  This inconsistency is confusing. Both should use the same approach.